File: src/main/java/rbq/wtf/lycoris/client/wrapper/wrappers/annotation/WrapObject.java
package rbq.wtf.lycoris.client.wrapper.wrappers.annotation;

import rbq.wtf.lycoris.client.wrapper.MapEnum;
import rbq.wtf.lycoris.client.wrapper.wrappers.annotation.repeat.WrapObjects;

import java.lang.annotation.*;

@Documented
@Target(ElementType.FIELD)
@Retention(RetentionPolicy.RUNTIME)
@Repeatable(WrapObjects.class)
/**
 * Auto setup field
 */
public @interface WrapObject {
    String mcpName();
    MapEnum targetMap();
    String customSrgName() default "none";
}

File: src/main/java/rbq/wtf/lycoris/client/wrapper/wrappers/annotation/WrapObject.kt
package rbq.wtf.lycoris.client.wrapper.wrappers.annotation

import rbq.wtf.lycoris.client.wrapper.MapEnum
import rbq.wtf.lycoris.client.wrapper.wrappers.annotation.repeat.WrapObjects

@MustBeDocumented
@Target(AnnotationTarget.FIELD)
@Retention(AnnotationRetention.RUNTIME)
@JvmRepeatable(
    WrapObjects::class
)
/**
 * Auto setup field
 */
annotation class WrapObject(val mcpName: String, val targetMap: MapEnum, val customSrgName: String = "none")